app/controllers/expenditures_controller.rb

```diff
@@ -0,0 +1,72 @@
+class ExpendituresController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_expenditure, only: %i[show edit update destroy]
+
+  # GET /expenditures or /expenditures.json
+  def index
+    @group_id = Group.find(params[:group_id])
+    @expenditures = Expenditure.where(group_id: @group_id.id)
+  end
+
+  # GET /expenditures/1 or /expenditures/1.json
+  def show; end
+
+  # GET /expenditures/new
+  def new
+    @group_id = Group.find(params[:group_id]).id
+    @expenditure = Expenditure.new
+  end
+
+  # GET /expenditures/1/edit
+  def edit; end
+
+  # POST /expenditures or /expenditures.json
+  def create
+    @expenditure = Expenditure.new(expenditure_params)
+    @expenditure.user_id = current_user.id
+    @group = Group.find(params[:group_id])
+
+    @expenditure.group_id = @group.id
+
+    if @expenditure.save
+      redirect_to user_group_expenditures_url(user_id: current_user.id, group_id: @group.id)
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] You are supposed to create success and fail paths for `create,` `update` and `destroy ` actions